x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/scalar/package-info.java

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+/**
+ * Functions that take a row of data and produce a row of data without holding
+ * any state between rows. This includes both the {@link org.elasticsearch.xpack.ql.expression.function.scalar.ScalarFunction}
+ * subclass to link into the QL infrastucture and the {@link org.elasticsearch.compute.operator.EvalOperator.ExpressionEvaluator}
+ * implementation to run the actual function.
+ *
+ * <h2>Guide to adding new function</h2>
+ * <p>
+ *     Adding functions is <strong>fairly</strong> easy and should be fun!
+ *     This is a step by step list of how to do it.
+ * </p>
+ * <ol>
+ *     <li>Fork the <a href="github.com/elastic/elasticsearch">Elasticsearch repo</a>.</li>
+ *     <li>Clone your fork locally.</li>
+ *     <li>Add Elastic's remote, it should look a little like:
+ *         {@code<pre>
```

Comment:
I don't think this is rendered correctly in the docs; the `<pre>` tag is printed verbatim and everything is squished into one line, going by your screenshot. I guess the tag needs to be outside the `{@code ... }` block?

Also applies in other sections below.

Reply:
> Also applies in other sections below.

Ah! I was mostly checking within intellij and just quickly eyeballed the screenshot. I'll fix.